bestiary/core/models.py

```diff
@@ -344,3 +374,37 @@ def save(self, *args, **kwargs):
         filters_str = json.dumps(self.filters, sort_keys=True)
         self.filters_hash = hashlib.sha1(filters_str.encode('utf-8')).hexdigest()
         super().save(*args, **kwargs)
+
+
+class Credential(EntityBase):
+    """Model class for users data source tokens.
+
+    This class is meant to represent the tokens associated to a user
+    that can be used against a specific data source.
+
+    Every credential object must have a `token` for a specific
+    `datasource` (`DataSourceType`) and must belong to a user.
+
+    :param user: User which the token belongs to.
+    :param name: Name for the token given by the user.
+    :param token: Token for the data source.
+    :param datasource: Datasource type which the token is for.
+    """
+    user = ForeignKey(settings.AUTH_USER_MODEL,
+                      on_delete=CASCADE,
+                      help_text='user allowed to user this token.')
+    name = CharField(max_length=MAX_SIZE_TITLE_FIELD,
+                     help_text='Name of the token.')
+    token = EncryptedField(max_length=MAX_SIZE_CHAR_INDEX,
```

Comment:
I think this is wrong. This is only a restriction for indices but token isn't going to be part of an index. This can be set to 256 chars without problems, so the full key is stored.